### PR TITLE
GSS BZ-2120944 Test Large Omap Objects FS Meta Data

### PR DESCRIPTION
**Pull request by OdedViner**
https://bugzilla.redhat.com/show_bug.cgi?id=2120944

### CODE REVIEW

File: tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            f"access_mode={constants.ACCESS_MODE_RWX}, volume_mode={constants.VOLUME_MODE_FILESYSTEM}"
+        )
+        self.pod_obj = pod_factory(
+            interface=constants.CEPHFILESYSTEM,
+            pvc=pvc_obj,
+            status=constants.STATUS_RUNNING,
+            pod_dict_path=pod_dict_path,
+            raw_block_pv=raw_block_pv,
+        )
+        self.pod_obj.exec_cmd_on_pod(
```
Comment (OdedViner) [inline tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py:66]: https://github.com/red-hat-storage/ocs-ci/blob/7627f289dcb1ad36b679b4431846dd3b660993e1/ocs_ci/ocs/resources/pod.py#L292

File: tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+)
+
+log = logging.getLogger(__name__)
+
+
+@brown_squad
+@tier2
+@ignore_leftovers
+@bugzilla("2120944")
+@skipif_external_mode
```
Comment (PrasadDesala) [inline tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py:24]: This test is not supported on external mode?

File: tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+@skipif_ocs_version("<4.12")
+@pytest.mark.polarion_id("OCS-XXXX")
```
Comment (PrasadDesala) [inline tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py:26]: Polarion id missing

File: tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+            interface=constants.CEPHFILESYSTEM,
+            size="20",
+        )
+        pod_dict_path = constants.NGINX_POD_YAML
+        raw_block_pv = False
+        log.info(
+            f"Created new pod sc_name={constants.CEPHFILESYSTEM} size=20Gi, "
+            f"access_mode={constants.ACCESS_MODE_RWX}, volume_mode={constants.VOLUME_MODE_FILESYSTEM}"
+        )
+        self.pod_obj = pod_factory(
```
Comment (PrasadDesala) [inline tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py:59]: any reason for not creating pvc using pod_factory?

File: tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+            interface=constants.CEPHFILESYSTEM,
+            pvc=pvc_obj,
+            status=constants.STATUS_RUNNING,
+            pod_dict_path=pod_dict_path,
+            raw_block_pv=raw_block_pv,
+        )
+        cmd = (
+            "mkdir -p dir3/.snap; for n in {1..100}; do touch dir3/file{0..11000}; "
```
Comment (PrasadDesala) [inline tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py:67]: the mount point of the ngnix pod is /var/lib/www/html, looks like you are not creating these on the mount path. Please check

File: tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py (hunk 2 above)
Comment (OdedViner) [reply tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py:24]: done

File: tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py (hunk 4 above)
Comment (OdedViner) [reply tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py:59]: I want to configure the PVC size

File: tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py (hunk 5 above)
Comment (OdedViner) [reply tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py:67]: done

File: tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py (hunk 3 above)
Comment (OdedViner) [reply tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py]: done

File: tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+            interface=constants.CEPHFILESYSTEM,
+            pvc=pvc_obj,
+            status=constants.STATUS_RUNNING,
+            pod_dict_path=pod_dict_path,
+            raw_block_pv=raw_block_pv,
+        )
+        cmd = (
+            "mkdir -p /var/lib/www/html/dir3/.snap; for n in {1..100}; do touch /var/lib/www/html/dir3/file{0..11000}; "
+            "mkdir -p /var/lib/www/html/dir3/.snap/snap_$n; rm -f /var/lib/www/html/dir3/file{0..11000}; done"
+        )
```
Comment (yitzhak12) [inline tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py:65]: Can you add a comment or a log message of what this command does?

Comment (OdedViner) [reply tests/manage/z_cluster/test_large_omap_objects_fs_meta_data.py:65]: done

**Review by PrasadDesala — APPROVED** [review]

**Review by yitzhak12 — DISMISSED** [review]

**Comment by OdedViner** [comment]
```2024-02-26 19:01:35  17:01:34 - MainThread - tests.functional.z_cluster.test_large_omap_objects_fs_meta_data - INFO  - Create large number of files under a directory exceeding the mds configuration mds_bal_split_size.
2024-02-26 19:01:35  17:01:34 - MainThread - ocs_ci.utility.utils - INFO  - Executing command: oc -n namespace-test-b085281fcd624c08afd7d76e4 exec pod-test-cephfs-bb7c2bcf51e54d1c825ce299 -- bash -c "mkdir -p /var/lib/www/html/dir3/.snap; for n in {1..100}; do touch /var/lib/www/html/dir3/file{0..11000}; mkdir -p /var/lib/www/html/dir3/.snap/snap_$n; rm -f /var/lib/www/html/dir3/file{0..11000}; done"
2024-02-26 19:01:35  17:01:35 - MainThread - ocs_ci.utility.utils - INFO  - searching for plugin: _n
2024-02-26 19:29:27  17:29:24 - MainThread - ocs_ci.utility.utils - WARNING  - Command stderr: mkdir: can't create directory '/var/lib/www/html/dir3/.snap/snap_1': Permission denied
2024-02-26 19:29:27  mkdir: can't create directory '/var/lib/www/html/dir3/.snap/snap_2': Permission denied
2024-02-26 19:29:27  mkdir: can't create directory '/var/lib/www/html/dir3/.snap/snap_3': Permission denied
2024-02-26 19:29:27  mkdir: can't create directory '/var/lib/www/html/dir3/.snap/snap_4': Permission denied```
 @PrasadDesala @vkathole 
The script ran ~28 min. 
1. Can we change the loop parameters so the test time will be shorter? What do you think?  
2. Do you understand the permission-denied warnings?

**Comment by OdedViner** [comment]
Automated by RHCS QE team  - https://github.com/red-hat-storage/cephci/blob/master/tests/cephfs/cephfs_bugs/test_large_omap.py

**Pull request closed without merging**